src/components/Table/Table.jsx

```diff
@@ -1,28 +1,85 @@
 import React, { Component } from 'react';
 import styles from './Table.module.css';
+import Pagination from '../Pagination/Pagination';
 
 const students = [
   { firstName: 'John', secondName: 'Johnson', birthYear: 1978 },
-  { firstName: 'John1', secondName: 'Johnson1', birthYear: 1988 },
-  { firstName: 'John2', secondName: 'Johnson2', birthYear: 1908 },
-  { firstName: 'John3', secondName: 'Johnson3', birthYear: 1998 },
-  { firstName: 'John4', secondName: 'Johnson4', birthYear: 1990 },
+  { firstName: 'Leanne', secondName: 'Graham', birthYear: 1988 },
+  { firstName: 'Ervin', secondName: 'Howell', birthYear: 1956 },
+  { firstName: 'Bauch', secondName: 'Clementine', birthYear: 2002 },
+  { firstName: 'Patricia', secondName: 'Labsak', birthYear: 1997 },
+  { firstName: 'Dennis', secondName: 'Schulz', birthYear: 1973 },
+  { firstName: 'Kasey', secondName: 'Neistat', birthYear: 1987 },
+  { firstName: 'Glenna', secondName: 'Reichert', birthYear: 2005 },
+  { firstName: 'Glenna', secondName: 'Reichert', birthYear: 2005 },
 ];
 
 class Table extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      students,
       page: 1,
-      itemPerPage: 2,
+      itemPerPage: 4,
+      students,
     };
   }
 
+  getVisibleStudents() {
+    const pages = Math.ceil(this.state.students.length / this.state.itemPerPage);
+    if (this.state.page > pages) {
+      this.setState({ page: pages });
+    }
+    const start = (this.state.page - 1) * this.state.itemPerPage;
+    const end = start + this.state.itemPerPage;
+    return students.slice(start, end);
+  }
+
+  updateItemsPerPage = (num) => {
+    this.setState({ itemPerPage: +num });
```

Comment:
it's old style usage setState
https://ru.reactjs.org/docs/react-component.html#setstate
here you can read new api this method with call back, it's almoust like this but litle be defferent